s3/pyproject.toml

```diff
@@ -0,0 +1,83 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+[tool.poetry]
+package-mode = false
+requires-poetry = ">=2.0.0"
+
+[tool.poetry.dependencies]
+python = "^3.10"
+ops = ">=2.18.0"
+pydantic = "^2.11.3"
```

Comment:
This overrides the pydeps of data_models cham lib. Most probably after https://github.com/canonical/data-platform-libs/pull/201 is merged the pydantic dep should be moved into the `charm-libs` group.